### PR TITLE
Use image index digest instead of plain image digest

### PR DESCRIPTION
**Pull request by StefanHauth**
## Description

Currently the image digest for AMD64 images is used in OneAgent daemonsets. This causes errors if OneAgents should be rolled out on nodes with architectures other than amd64.

Instead of using the digest of the amd64 image the operator shall use the digest of the image index, that points to an image manifest containing images for all supported architectures.

## How can this be tested?

- setup a K8s cluster with arm and amd nodes
- build and deploy a multi-arch operator
- create a dynakube with proper tolerations for one agent pods:
```
       tolerations:
        - effect: NoSchedule
          key: kubernetes.io/arch
          value: arm64
          operator: Equal
```
- deploy DynaKube
- check that OneAgents can properly start on arm nodes

## Checklist

- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly with a single label
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: pkg/oci/registry/client.go [hunk 1]
```diff
@@ -116,10 +116,21 @@ func (c *Client) GetImageVersion(ctx context.Context, imageName string) (ImageVe
 	if err != nil {
 		return ImageVersion{}, fmt.Errorf("descriptor.Image(): %w", err)
 	}
-	dig, err := img.Digest()
+
+	// use image digest as a fallback
+	digestFn := img.Digest
```
Comment (luhi-DT) [inline pkg/oci/registry/client.go:121]: was it on purpose to not use the Digest() function?

File: pkg/oci/registry/client.go [hunk 2]
```diff
@@ -116,10 +116,21 @@ func (c *Client) GetImageVersion(ctx context.Context, imageName string) (ImageVe
...
+
+	// try to get image manifest to cover multi arch images
+	digestSrc, err := descriptor.ImageIndex()
```
Comment (luhi-DT) [inline pkg/oci/registry/client.go:124]: you are re-assigning the error variable here that means the error check in 130 does not make sense anymore or am I missing something

File: pkg/oci/registry/client.go (hunk 1 above)
Comment (StefanHauth) [reply pkg/oci/registry/client.go:121]: I'm using a function pointer here that is later on used to call the Digest() function. Depending on the outcome of line 124, the function pointer might be switched to a different Digest() function. I tried to avoid lengthy if... else constructs here. Alternative implementation:
```
	var dig containerv1.Hash
	// try to get image manifest to cover multi arch images
	if imgIdx, err := descriptor.ImageIndex(); err == nil {
		dig, err = imgIdx.Digest()
		if err != nil {
			return ImageVersion{}, fmt.Errorf("img.Digest(): %w", err)
		}
	} else {
		dig, err = img.Digest()
		if err != nil {
			return ImageVersion{}, fmt.Errorf("img.Digest(): %w", err)
		}
	}
```

File: pkg/oci/registry/client.go (hunk 2 above)
Comment (StefanHauth) [reply pkg/oci/registry/client.go:124]: The error check in 130 is for the outcome of 129, Or am I understanding something wrong?

Comment (0sewa0) [reply pkg/oci/registry/client.go:124]: we do not care about the error from `descriptor.ImageIndex()`, as that errors (in theory) when not a image-index was given but an actual image (like `snapshot-amd64` instead of just `snapshot`)

the actual error that we care about, and should be propagated,  is what the `digestFn` would return, as that shouldn't error

TLDR: This is fine, BUT pls rename `digestSrc` to `imageIndex` as `digestSrc` is veeery generic 😅

**Review by 0sewa0 — APPROVED** [review]
